internal/ssm/install.go

```diff
@@ -34,21 +80,79 @@ type PkgSource interface {
 }
 
 func Install(ctx context.Context, tracker *tracker.Tracker, source Source) error {
+	if err := downloadFileWithRetries(ctx, source); err != nil {
+		return errors.Wrap(err, "failed to install ssm installer")
+	}
+	if err := runInstallWithRetries(ctx); err != nil {
+		return errors.Wrapf(err, "failed to install ssm agent")
+	}
+
+	return tracker.Add(artifact.Ssm)
+}
+
+func downloadFileWithRetries(ctx context.Context, source Source) error {
+	// Retry up to 3 times to download and validate the signature of
+	// the SSM setup cli.
+	var err error
+	for i := 0; i < 3; i++ {
+		err = downloadFile(ctx, source)
+		if err == nil {
+			break
+		}
+	}
+	return err
+}
+
+func downloadFile(ctx context.Context, source Source) error {
 	installer, err := source.GetSSMInstaller(ctx)
 	if err != nil {
 		return err
 	}
 	defer installer.Close()
 
-	if err := artifact.InstallFile(installerPath, installer, 0o755); err != nil {
-		return errors.Wrap(err, "failed to install ssm installer")
+	signature, err := source.GetSSMInstallerSignature(ctx)
+	if err != nil {
+		return err
 	}
+	defer signature.Close()
 
-	if err = runInstallWithRetries(ctx); err != nil {
-		return errors.Wrapf(err, "failed to install ssm agent")
+	var installerBuffer bytes.Buffer
+	installerTee := io.TeeReader(installer, &installerBuffer)
+
+	if err := validateSetupSignature(installerTee, signature); err != nil {
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("validating ssm-setup-cli signature: %w, err)
```

i think here the context is super important, if not, we are just gonna get the error from the pgp library without telling us what's the signature being valdating